e2e/_suites/fleet/features/fleet_mode_agent.feature

```diff
@@ -40,6 +40,17 @@ Examples:
 | centos |
 | debian |
 
+@upgrade-agent
+Scenario Outline: Upgrading the installed <os> agent
+  Given a "<os>" agent "stale" is deployed to Fleet with "tar" installer
+    And certs for "<os>" are installed
+  When agent is upgraded to version "latest"
+  Then process waits for "2m"
+    And agent is in version "latest"
```

Comment:
this is great - after this I would like to see some of the other validation steps copied in, to make sure it is sending docs in to ES.  ... and I realize we may not have that for Fleet cases (only for stand-alone) so we may leave this and add it as a separate item if we must.

Reply:
its a really weak validation as it is, no proof of what Agent is or can do after it is upgraded.  But it is a start, nothing wrong with incremental improvements